test-schema.json

```diff
@@ -27,6 +27,26 @@
                 "type": "array",
                 "items": { "$ref": "#/$defs/test" },
                 "minItems": 1
+            },
+            "specification":{
+                "description": "A reference to a specification document which defines the behavior tested by this test case. Typically this should be a JSON Schema specification document, though in cases where the JSON Schema specification points to another RFC it should contain *both* the portion of the JSON Schema specification which indicates what RFC (and section) to follow as *well* as information on where in that specification the behavior is specified.",
+
+                "type": "array",
+                "minItems": 1,
+                "items":{
+                    "properties": {
```

Comment:
I think all the values here will be (should be) strings, so probably you can/should use `propertyNames` here (probably with a `oneOf` + `enum` + `pattern`).

Reply:
should I leave properties as it is or put everything inside additionalProperties?
perl5 and quote will take just strings, and having them inside additionalProperties implies using `oneOf: type:string, type:string+pattern`